packages/components/src/hooks/use-hover.ts

```diff
@@ -15,32 +13,26 @@ export const useHover = <T extends HTMLElement | SVGSVGElement>(
 
     React.useEffect(() => {
         const node = ref.current;
-
-        if (!node) return;
-
-        const handleAddEvents = () => {
-            if (isMobileOs()) {
-                node.addEventListener('touchstart', handleHoverBegin);
-                node.addEventListener('touchend', handleHoverFinish);
+        if (node) {
+            if (should_prevent_bubbling) {
+                node.addEventListener('mouseenter', handleHoverBegin);
+                node.addEventListener('mouseleave', handleHoverFinish);
             } else {
-                node.addEventListener(should_prevent_bubbling ? 'mouseenter' : 'mouseover', handleHoverBegin);
-                node.addEventListener(should_prevent_bubbling ? 'mouseleave' : 'mouseout', handleHoverFinish);
+                node.addEventListener('mouseover', handleHoverBegin);
+                node.addEventListener('mouseout', handleHoverFinish);
             }
-        };
-
-        const handleRemoveEvents = () => {
-            if (isMobileOs()) {
-                node.removeEventListener('touchstart', handleHoverBegin);
-                node.removeEventListener('touchend', handleHoverFinish);
-            } else {
-                node.removeEventListener(should_prevent_bubbling ? 'mouseenter' : 'mouseover', handleHoverBegin);
-                node.removeEventListener(should_prevent_bubbling ? 'mouseleave' : 'mouseout', handleHoverFinish);
-            }
-        };
-
-        handleAddEvents();
 
-        return handleRemoveEvents;
+            return () => {
+                if (should_prevent_bubbling) {
+                    node.removeEventListener('mouseenter', handleHoverBegin);
+                    node.removeEventListener('mouseleave', handleHoverFinish);
+                } else {
+                    node.removeEventListener('mouseover', handleHoverBegin);
+                    node.removeEventListener('mouseout', handleHoverFinish);
+                }
+            };
+        }
+        return undefined;
```

Comment:
reversion of changes from https://github.com/binary-com/deriv-app/pull/9562

Reply:
This is fine. 
@suisin-deriv Please take note. The touch events handler should be on its own hook. It should not be related to the use-hover hook.